### PR TITLE
Update heads and fix all test failures

### PR DESCRIPTION
**Pull request by dzhulgakov**
All dependent commits in pytorch/onnx has been landed already.

Stuff that broke over 2 weeks:
1) addition of stacktrace - filter it out everywhere
2) changes to ONNX that removed dilation on MaxPool and renamed Caffe2ConvTranspose and SpatialBN
3) Random renumbering in tests - not sure why they happen
4) Addition of types in ONNX IR - all "expected" tests has changed
5) Some internal functions in pytorch (like _flatten) got broken
also, already landed:
6) Fix for https://github.com/pytorch/pytorch/commit/621fbd5c4e4cae0e80c659c4120dcb673bbda8af#diff-675195b142afaf316f8130c44f5388c5 that broke BatchNorm

Concerning - test_conv did change the output blob, I didn't look much why exactly.

There are still warnings in the output but they don't seem to break stuff:
```
onnx-pytorch/pytorch/torch/onnx/__init__.py:243: UserWarning: ONNX export failed on clone because torch.onnx.symbolic.clone does not exist
```

### CODE REVIEW

**Comment by houseroad** [comment]
That UserWarning means PyTorch tries to export some operators called Clone. Do we have clone operator?

**Review by zdevito — APPROVED** [review]
Looks good.

The number change around because any time you create a new node it gets a new number. Depending on how many optimization passes you do on the graph you may get different final numbers. Renumbering when printing the graph is possible but it is a tradeoff: you get more consistent number in a graph, but you lose the ability to print single nodes when debugging.

**Review by houseroad — APPROVED** [review]
Everything looks good. IIUC, clone op is generated when we try to make tensor contiguous.
So probably, we can have an identity symbolic function for clone node. Or we can eliminate clone nodes in ToONNX.

**Comment by dzhulgakov** [comment]
Hm, build fails on `/home/jenkins/buildbot/workspace/onnx-builds/onnx-pytorch-PR-py2-linux/pytorch/aten/src/ATen/native/cuda/NativeFunctionsCuda.cu:1:34: fatal error: ATen/NativeFunctions.h: No such file or directory` even though everything seems to work locally for me.
